src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -150,18 +282,128 @@ public class PipelineCacheServer
             return pipelineCacheClient;
         }
 
-        private Task DownloadPipelineCacheAsync(
+        private async Task DownloadPipelineCacheAsync(
+            AgentTaskPluginExecutionContext context,
             DedupManifestArtifactClient dedupManifestClient,
             DedupIdentifier manifestId,
             string targetDirectory,
+            bool isTar,
+            string contentFormat,
             CancellationToken cancellationToken)
         {
-            DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
-                manifestId,
-                targetDirectory,
-                proxyUri: null,
-                minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+
+            // Throw for bunch of invalid combinations of istar and contentFormat.
```

Comment:
Even for restore, we need the environment variable to set, yes?
So, we will have to throw for scenarios where the variable is set, but the upload was not tar, and vice-versa?

Reply:
So, the decision is, on upload, check the environment variable, but DO NOT modify the existing key. On download - based on the content format, download whether it's tar or not.